### ISSUE TITLE
Does Polyform Shield prohibit forking software projects?

### ISSUE THREAD

**Issue by emidoots** [polyformproject/polyform-licenses#84]
Hello, very interesting project (and authors!) here, thank you for making this available.

I am looking at adopting PolyForm Shield for a project of mine, as it appears to be the most liberal and developer-friendly (read: Apache/MIT-like) license that exists, while still prohibiting someone from launching a competing SaaS version of my project.

I am not a lawyer, and I am not trying to ask for free legal advice (sorry if it comes across that way), but based on my interpretation of the PolyForm Shield license the definition of the noncompete seems ambiguous to me.

> Noncompete
>
> Any purpose is a permitted purpose, except for providing any **product** that competes with the software or any **product** the licensor or any of its affiliates provides using the software.
>
> [...] **Goods and services compete even when provided free of charge**. If you **market a product** as a practical substitute for the software or another product, it definitely competes.

Basically my questions are:

* Does a mere fork of my project on GitHub, with no modifications, count as the person who forked it now offering a competing **product** free of charge?
* What if that fork had substantial changes, thus becoming a free (no cost) competitor (layman's terms) maintained by the community?

My hope is that the answer would be these _do not count as products_ - but I fear they do, in which case PolyForm does not offer a license liberal enough for my use case as I want to ensure I explicitly allow for these two properties to be permitted.

Appreciate any thoughts you have!

**Comment by kemitchell** [comment]
@slimsag, I think I follow. But to be sure, could I ask you to describe what you want to allow that you think Shield prohibits? Maybe in one sentence?

**Comment by emidoots** [comment]
**The ability to redistribute (fork) a Shield-licensed project with no changes or major changes, as a community-driven not for profit competitor.**

- I make one small bug fix to a Shield-licensed project and offer that project to others
- I make a large series of bug fixes / changes to a Shield-licensed project and offer that project to others.

I want to allow the above as they are basic properties of developer-friendly projects, but I believe Shield prohibits both because that _is_ offering a competitor product at zero cost.

**Comment by kemitchell** [comment]
The shield and perimeter licenses are largely "defensive" licenses. The use case we had in mind was company-driver projects.

The licenses don't anticipate, or govern, outside contribution back to the company's project. The Changes and New Works License sections address the kinds of customizations and fixes that users are likely to make. But they don't discuss submitting those changes back to the vendor or how they might be licensed.

We might put more thought on how to welcome contributions back. But I don't think the primary use case for these licenses will want full "forkability"---the ability to start and maintain a version different from the "official". That said, there's nothing stopping Shield project developers from allowing and inviting patches back as-is.
